service/src/main/resources/db/migration/R__dev_temp.sql

```diff
@@ -19,33 +19,36 @@ CREATE TABLE vasu_template(
 
 CREATE TRIGGER set_timestamp BEFORE UPDATE ON vasu_template FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
 
-CREATE TABLE vasu_content(
+CREATE TABLE vasu_document(
     id uuid PRIMARY KEY DEFAULT ext.uuid_generate_v1mc(),
     created timestamp with time zone DEFAULT now() NOT NULL,
     updated timestamp with time zone DEFAULT now() NOT NULL,
-    content jsonb NOT NULL,
-    authors_content jsonb NOT NULL,
-    vasu_discussion_content jsonb NOT NULL,
-    evaluation_discussion_content jsonb NOT NULL
+    child_id uuid NOT NULL REFERENCES child(id),
+    template_id uuid NOT NULL REFERENCES vasu_template(id),
+    modified_at timestamp with time zone NOT NULL
 );
 
-CREATE TRIGGER set_timestamp BEFORE UPDATE ON vasu_content FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
+CREATE TRIGGER set_timestamp BEFORE UPDATE ON vasu_document FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
 
-CREATE TABLE vasu_document(
+CREATE INDEX idx$vasu_document_child_id ON vasu_document(child_id);
+CREATE INDEX idx$vasu_document_template_id ON vasu_document(template_id);
+
+CREATE TABLE vasu_content(
     id uuid PRIMARY KEY DEFAULT ext.uuid_generate_v1mc(),
     created timestamp with time zone DEFAULT now() NOT NULL,
     updated timestamp with time zone DEFAULT now() NOT NULL,
-    child_id uuid NOT NULL REFERENCES child(id),
-    template_id uuid NOT NULL REFERENCES vasu_template(id),
-    content_id uuid NOT NULL REFERENCES vasu_content(id),
-    modified_at timestamp with time zone NOT NULL
+    document_id uuid NOT NULL REFERENCES vasu_document(id),
+    published_revision bigint DEFAULT NULL,
+    content jsonb NOT NULL,
+    authors_content jsonb NOT NULL,
+    vasu_discussion_content jsonb NOT NULL,
+    evaluation_discussion_content jsonb NOT NULL
 );
 
-CREATE TRIGGER set_timestamp BEFORE UPDATE ON vasu_document FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
+CREATE TRIGGER set_timestamp BEFORE UPDATE ON vasu_content FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
 
-CREATE INDEX fk$vasu_document_child_id ON vasu_document(child_id);
-CREATE INDEX fk$vasu_document_template_id ON vasu_document(template_id);
-CREATE INDEX fk$vasu_document_content_id ON vasu_document(content_id);
+CREATE UNIQUE INDEX uniq$vasu_content_document_id_master ON vasu_content(document_id, (published_revision IS NULL)) WHERE published_revision IS NULL;
```

Comment:
is this the best way to ensure only one row per document with null value? borrowed this from internet but maybe it could be even simpler?

Reply:
would this work as well?
`CREATE UNIQUE INDEX uniq$vasu_content_document_id_master ON vasu_content(document_id) WHERE published_revision IS NULL;`

Reply:
Yeah I think that's enough and there's no need to duplicate the condition in the index itself...a very similar kind of partial index is used with `application_form` latest flag